src/librustdoc/html/static/main.js

```diff
@@ -374,11 +374,10 @@
                 results = {},
                 split = valLower.split("::");
 
-            // remove empty keywords
-            for (var j = 0; j < split.length; ++j) {
-                split[j].toLowerCase();
-                if (split[j] === "") {
-                    split.splice(j, 1);
+            for (var z = 0; z < split.length; ++z) {
+                if (split[z] === "") {
+                    split.slice(z, 1);
```

Comment:
Slice returns a new array, so this is a no-op. Did you want to .splice() instead? From the removed code side, it looks like the answer is yes.

More idiomatically, you can also push those that raen't the empty string.

Reply:
```
> x = [1,2,3];
(3) [1, 2, 3]
> x.splice(0, 1)
[1]
> x
(2) [2, 3]
```

So no, it clearly removes an entry from the array.

Reply:
Oh damn, you're absolutely right. Didn't see the missing 'p'!